python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -58,9 +58,150 @@ class MakeCoaddTempExpConfig(CoaddBaseTask.ConfigClass):
+    @link lsst.ip.diffim.modelPsfMatch.ModelPsfMatchTask ModelPsfMatchTask @endlink
+    is responsible for the PSF-Matching, and its config is accessed via `config.warpAndPsfMatch.psfMatch`.
+    The optimal configuration depends on aspects of dataset: the pixel scale, average PSF FWHM and
+    dimensions of the PSF kernel. These configs include the matching kernel size, the requested model PSF,
+    padding of the science PSF thumbnail and spatial sampling frequency of the PSF.
+    The following list presents the most common problems that arise from a misconfigured ModelPsfMatchTask
+    and corresponding solutions. All assume the default Alard-Lupton kernel, with configs accessed via
+    ```config.warpAndPsfMatch.psfMatch.kernel['AL']```.
+
+    *Troublshooting PSF-Matching Configuration:*
+    - Matched PSFs look boxy: The matching kernel is too small. _Increase the matching kernel size.
+        For example:_
+
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].kernelSize=27  # default 21
+
+        Note that increasing the kernel size also increases runtime.
+    - Matched PSFs look ugly (dipoles, quadropoles, donuts): unable to find good solution
+        for matching kernel. _Provide the matcher with more data by either increasing
+        the spatial sampling by decreasing the spatial cell size,_
+
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].sizeCellX = 64  # default 128
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].sizeCellY = 64  # default 128
+
+        _or increasing the padding around the Science PSF, for example:_
+
+            config.warpAndPsfMatch.psfMatch.autoPadPsfTo=1.6  # default 1.4
+
+        Optionally, for debugging the effects of padding, level of padding may be manually
+        controlled by setting:
+
+            config.warpAndPsfMatch.psfMatch.doAutoPadPsf = False  # default True
+            config.warpAndPsfMatch.psfMatch.padPsfBy = 6  # pixels. default 0
+
+    - Decorrelation noise: Matching a large PSF to a smaller PSF produces
+        a decorrelated noise pattern.
+        _Increase the size of the requested model PSF. For example:_
+
+            config.modelPsf.defaultFwhm = 11  # Gaussian sigma in units of pixels.
+
+    - High frequency (sometimes checkered) noise: The matching basis functions are too small.
+        _Increase the width of the Gaussian basis functions. For example:_
+
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].alardSigGauss=[1.5, 3.0, 6.0]
+            # from default [0.7, 1.5, 3.0]
+
+
+    @section pipe_tasks_makeCoaddTempExp_Debug  Debug variables
+
+    MakeCoaddTempExpTask has no debug output, but its subtasks do.
+
+    @section pipe_tasks_makeCoaddTempExp_Example   A complete example of using MakeCoaddTempExpTask
+
+    This example uses the packac ci_hsc to show how MakeCoaddTempExp fits into the larger Data Release Processing.
+    Set up by running:
```

Comment:
Nice to have an example! Going to be tricky keeping this up-to-date if anything changes.